### ISSUE TITLE
Inaccuracies in Test Results Using Provided Network Weights

### ISSUE THREAD

**Issue by xy1999729** [zjuluolun/BEVPlace#7]
I would like to express my gratitude for sharing your research and data openly. However, I have encountered an issue that I believe warrants discussion. While conducting tests using the provided network weights on the dataset you shared, I observed that the achieved accuracy did not align with the precision levels stated in your paper. This discrepancy has prompted me to seek further clarification and insight into potential factors contributing to this divergence.
The detail results are as follow:
KITTI     | 00       | 02       | 05      | 06    | Mean
paper    | 99.7    | 98.1    | 99.3    | 100  | 99.275
my test | 98.53  | 95.16  | 98.88 | 100  | 98.1425

**Comment by zjuluolun** [comment]
Thanks for your interest! I also checked the codes and found that the accuracy did not align with the precision levels stated in the paper. This is because the current pre-trained checkpoint is not the latest and there are also some bugs in the bev generation script. Sorry for the confusion and we will update our repo to the latest version ASAP.

**Comment by xy1999729** [comment]
> @xy1999729 In the previous version, the provided weights were trained using images generated by a different BEV generation script. We have now uploaded the correct checkpoint and addressed several bugs. Please feel free to ask me if you have any further questions.

Thank you for your inspiring work and timely response. I will try your latest version. If possible, could you share your training script? I would greatly appreciate it.

**Comment by zjuluolun** [comment]
@xy1999729 Finally... I found the correct checkpoint and dataset in some corner... You can run main.py to reproduce our numbers in the paper now. The newly uploaded dataset contains BEV images. These images are generated by firstly cropping the point clouds and downsampling the clouds into 4096 points. The provided BEV generation script omits the step of downsampling points. This is one reason for the performance degeneration.  In addition, we use the ground truth poses of LiDAR for evaluation on KITTI. A minor decrease in recalls may occur if you use the raw pose files (of cameras). Anyway, things are going right now.
       In our updated repo, there are two checkpoints called "checkpoint_paper_kitti" and "checkpoint_custom_bev". I took the trouble to do this since I lost the original BEV generation script...  So I rewrote the script and trained a new model. If you have any custom data for evaluation, please use the "checkpoint_custom_bev" checkpoint. 
      We'll release the training scripts of ResNet+NetVLAD on the KITTI and the benchmark dataset soon.  I hope this helps you to some extent.  (The ResNet+NetVLAD network archives comparable performance to BEVPlace on the benchmark dataset. )

**Comment by xy1999729** [comment]
> @xy1999729 Finally... I found the correct checkpoint and dataset in some corner... You can run main.py to reproduce our numbers in the paper now. The newly uploaded dataset contains BEV images. These images are generated by firstly cropping the point clouds and downsampling the clouds into 4096 points. The provided BEV generation script omits the step of downsampling points. This is one reason for the performance degeneration. In addition, we use the ground truth poses of LiDAR for evaluation on KITTI. A minor decrease in recalls may occur if you use the raw pose files (of cameras). Anyway, things are going right now. In our updated repo, there are two checkpoints called "checkpoint_paper_kitti" and "checkpoint_custom_bev". I took the trouble to do this since I lost the original BEV generation script... So I rewrote the script and trained a new model. If you have any custom data for evaluation, please use the "checkpoint_custom_bev" checkpoint. We'll release the training scripts of ResNet+NetVLAD on the KITTI and the benchmark dataset soon. I hope this helps you to some extent. (The ResNet+NetVLAD network archives comparable performance to BEVPlace on the benchmark dataset. )

Thank you for your contributions to the community. Your kindly responses have been immensely helpful to me and of course all the follower. Regarding the "custom_bev" weights, as you stated in the paper, training with the KITTI00 dataset led to outstanding generalization performance. How does this "custom_bev" weight differ from the previous ones?

**Comment by zjuluolun** [comment]
> > @xy1999729 Finally... I found the correct checkpoint and dataset in some corner... You can run main.py to reproduce our numbers in the paper now. The newly uploaded dataset contains BEV images. These images are generated by firstly cropping the point clouds and downsampling the clouds into 4096 points. The provided BEV generation script omits the step of downsampling points. This is one reason for the performance degeneration. In addition, we use the ground truth poses of LiDAR for evaluation on KITTI. A minor decrease in recalls may occur if you use the raw pose files (of cameras). Anyway, things are going right now. In our updated repo, there are two checkpoints called "checkpoint_paper_kitti" and "checkpoint_custom_bev". I took the trouble to do this since I lost the original BEV generation script... So I rewrote the script and trained a new model. If you have any custom data for evaluation, please use the "checkpoint_custom_bev" checkpoint. We'll release the training scripts of ResNet+NetVLAD on the KITTI and the benchmark dataset soon. I hope this helps you to some extent. (The ResNet+NetVLAD network archives comparable performance to BEVPlace on the benchmark dataset. )
> 
> Thank you for your contributions to the community. Your kindly responses have been immensely helpful to me and of course all the follower. Regarding the "custom_bev" weights, as you stated in the paper, training with the KITTI00 dataset led to outstanding generalization performance. How does this "custom_bev" weight differ from the previous ones?

These two weights are trained in the same way using scans 0-3000 of Seq. 00 of KITTI. The difference is that they use BEV images generated by different scripts.

**Comment by xy1999729** [comment]
> > > @xy1999729 Finally... I found the correct checkpoint and dataset in some corner... You can run main.py to reproduce our numbers in the paper now. The newly uploaded dataset contains BEV images. These images are generated by firstly cropping the point clouds and downsampling the clouds into 4096 points. The provided BEV generation script omits the step of downsampling points. This is one reason for the performance degeneration. In addition, we use the ground truth poses of LiDAR for evaluation on KITTI. A minor decrease in recalls may occur if you use the raw pose files (of cameras). Anyway, things are going right now. In our updated repo, there are two checkpoints called "checkpoint_paper_kitti" and "checkpoint_custom_bev". I took the trouble to do this since I lost the original BEV generation script... So I rewrote the script and trained a new model. If you have any custom data for evaluation, please use the "checkpoint_custom_bev" checkpoint. We'll release the training scripts of ResNet+NetVLAD on the KITTI and the benchmark dataset soon. I hope this helps you to some extent. (The ResNet+NetVLAD network archives comparable performance to BEVPlace on the benchmark dataset. )
> > 
> > 
> > Thank you for your contributions to the community. Your kindly responses have been immensely helpful to me and of course all the follower. Regarding the "custom_bev" weights, as you stated in the paper, training with the KITTI00 dataset led to outstanding generalization performance. How does this "custom_bev" weight differ from the previous ones?
> 
> These two weights are trained in the same way using scans 0-3000 of Seq. 00 of KITTI. The difference is that they use BEV images generated by different scripts.

Dear Luo, could you share your strategy to sample the positive and negative samples of the query?

**Comment by ANXPAX** [comment]
Dear Luo can you please provide training script

**Comment by zjuluolun** [comment]
Hi all, a newer version of BEVPlace, BEVPlace++ has been released. Please refer to the paper and codes for more details.
